components_library/boards/lupv_board.py

```diff
@@ -95,28 +97,39 @@ def __init__(
         # PLIC
         self.pic = Plic(pio_addr=0xc000000)
 
+        # LUPIO PIC
+        self.lupio_pic = LupioPIC(pio_addr=0x20002000)
+
         # LUPIO RNG
         self.lupio_rng = LupioRNG(pio_addr=0x20005000)
 
         # LUPIO RTC
         self.lupio_rtc = LupioRTC(pio_addr=0x20004000)
 
+        # LUPIO TMR
+        self.lupio_tmr = LupioTMR(pio_addr=0x20006000)
+
         # LUPIO TTY
-        self.lupio_tty = LupioTTY(pio_addr=0x20007000, pic = self.pic)
+        self.lupio_tty = LupioTTY(pio_addr=0x20007000, pic = self.lupio_pic)
         self.terminal = Terminal()
 
         # LUPIO BLK
-        self.lupio_blk = LupioBLK(pio_addr=0x20000000, pic = self.pic)
+        self.lupio_blk = LupioBLK(pio_addr=0x20000000, pic = self.lupio_pic)
 
         # Note: This only works with single threaded cores.
         self.pic.n_contexts = self.processor.get_num_cores() * 2
```

Comment:
Actually, now that our LupioPIC is in charge of the supervisor-level external interrupts, the PLIC should be left in charge of the machine-level external interrupts only. So technically, the number of contexts should only be `self.processor.get_num_cores()`.